pkgs/tools/networking/zerotierone/default.nix

```
@@ -65,6 +65,19 @@ in stdenv.mkDerivation {
 
   enableParallelBuilding = true;
 
+  # Ensure Rust compiles for the right target
+  env.CARGO_BUILD_TARGET = stdenv.hostPlatform.rust.rustcTarget;
+
+  # Cargo won't compile to target/release but to target/<RUST_TARGET>/release when a target is
+  # explicitly defined. The build-system however expects target/release. Hence we just symlink from
+  # the latter to the former.
+  preBuild = ''
+    mkdir --parent -- rustybits/target/release
```

```suggestion
    mkdir -p rustybits/target/release
```
Those args are well known and van be shortened

---

@wolfgangwalther Yes, this particular instance of flags is well understood, and I do not doubt that. Thank you for sharing your thoughts!

My point is, there are flags which clearly are not well understood. Consequently, there are flags which are somewhere in-between: not widely understood, but also not too exotic. Drawing this line (to shorten or not to shorten) can be hard. You provide a metric (and I would guess the specific criterion would be "which variant is more often found in nixpkgs"). However, I refuse to accept the majority case as the desirable state _just because it is the majority case_.

Consider the following example:

```bash
for i in $(find . -name \*.txt); do
    process "$i"
done
```

is [considered an anti-pattern, as it breaks on whitespace](https://stackoverflow.com/a/9612232). The canonical way likely would be akin this:

```bash
find . -name '*.txt' -print0 | 
    while IFS= read -r -d '' line; do 
        process "$line"
    done
```

Now lets see which one is more prevalent in the repo:

```console
$ grep -R 'for.*in.*\$(find' * | wc -l
175
$ grep -R 'while IFS= read' | wc -l
64
```

Now, the worse variant (potentially prone to breakage on whitespace) is more prevalent. Is this dramatic? No for sure not. Is it optimal? Also no, its possible to do better without much research.

What I want to drive home here is: knowing when shortening is fine, and when it might cause problems down the line is not easy. Finding a clear policy of when to do what is not easy. Majority vote can be quite a good metric to know if a pattern is common or not. Despite that, being more verbose is on the safer side, and the cost of doing so is little. So I prefer being on the more cautionary, more verbose side of things.